From: Javier Viguera <javier.viguera@digi.com>
Date: Fri, 11 May 2012 13:15:07 +0200
Subject: [PATCH] gplay_repeat

Fix some issues related with 'repeat' command in gplay.

1. repeat_current not working if the item was the last one of the
   playlist.

2. Running 'play previous [<]' does not honour 'repeat_current' mode and
   jumps to previous item in the playlist anyway.

CCORE_MX53_EXTENSIONS-140

Signed-off-by: Javier Viguera <javier.viguera@digi.com>
---
 tools/gplay/mfw_gplay.c |    3 +++
 1 file changed, 3 insertions(+)

diff --git a/tools/gplay/mfw_gplay.c b/tools/gplay/mfw_gplay.c
index de11e68..5e5a6f1 100755
--- a/tools/gplay/mfw_gplay.c
+++ b/tools/gplay/mfw_gplay.c
@@ -151,6 +151,7 @@ PlayItem * playlist_next(fsl_player_handle handle, options* opt)
         }
         case FSL_PLAYER_REPEAT_CURRENT:
         {
+            next = opt->current;
             pplayer->klass->stop(pplayer);
             //pplayer->klass->set_media_location(pplayer, opt->current->name, &drm_format);
             pplayer->klass->play(pplayer);
@@ -172,6 +173,7 @@ fsl_player_ret_val playlist_previous(fsl_player_handle handle, options* opt)
     PlayItem * current = opt->current;
     PlayItem * next = getPrevItem(current);
 
+#if 0
     if(next)
     {
         opt->current = next;
@@ -180,6 +182,7 @@ fsl_player_ret_val playlist_previous(fsl_player_handle handle, options* opt)
         pplayer->klass->set_media_location(pplayer, opt->current->name, &drm_format);
         pplayer->klass->play(pplayer);
     }
+#endif
     switch( opt->repeat )
     {
         case FSL_PLAYER_REPEAT_NONE:
